src/components/DocumentViewer.tsx

```diff
@@ -1,33 +1,39 @@
 import React, { useEffect, useRef } from 'react';
 import useInstance from '../context';
 import WebViewer from '@pdftron/webviewer';
-import type { WebViewerOptions } from '@pdftron/webviewer';
+import type { WebViewerOptions, WebViewerInstance } from '@pdftron/webviewer';
 import PropTypes from 'prop-types';
 
 export type TProps = {
   className?: string
+  onReady?: (instance: WebViewerInstance) => void
 } & WebViewerOptions
 
 export type TRef = React.RefObject<HTMLDivElement>
 
 const DocumentViewer = React.forwardRef<TRef, TProps>(
-  ({ className, ...rest }, ref) => {
+  ({ onReady, className, ...rest }, ref) => {
     const localRef = useRef<HTMLDivElement>(null);
     const { instance, setInstance } = useInstance();
 
     useEffect(() => {
       if (!instance) 
         WebViewer(rest, localRef.current as HTMLDivElement).then(ins => {
+          if (onReady) ins.Core.documentViewer.addEventListener('documentLoaded', () => {
+            onReady(ins);
+          }, { once: true });
```

Comment:
Here just call `onReady` without listening to documentLoaded event (remove the event listener)
```
.then(ins => {
 onReady?.(ins)
}
```